### PR TITLE
Fix parsing behavior of text-shadow property

### PR DESCRIPTION
**Pull request by canova**
Blur radius should reject negative values. There was already `Shadow` struct for properties like this. `filter: drop-shadow` also has the same syntax with that property. I thought sharing the same code would be better and used Shadow struct for parsing. Converted to SpecifiedTextShadow to get rid of unneccessary fields and to be able to convert its computed value. Maybe it would be better to avoid using `Shadow` or just using `Shadow` but sharing code and avoiding unneccessary fields seems better.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16754)

### CODE REVIEW

File: components/style/properties/longhand/inherited_text.mako.rs [hunk 1]
```diff
@@ -773,60 +785,59 @@ ${helpers.single_keyword("text-align-last",
         fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
             let mut iter = self.0.iter();
             if let Some(shadow) = iter.next() {
-                try!(shadow.to_css(dest));
+                shadow.to_css(dest)?;
```
Comment (emilio) [inline components/style/properties/longhand/inherited_text.mako.rs]: Given you're refactoring all this, why not:

```rust
match iter.next() {
    Some(shadow) => shadow.to_css(dest)?,
    None => return dest.write_str("none"),
}
// ...
```

File: components/style/properties/longhand/inherited_text.mako.rs [hunk 2]
```diff
@@ -750,6 +751,17 @@ ${helpers.single_keyword("text-align-last",
         pub color: Option<specified::CSSColor>,
     }
 
+    impl From<Shadow> for SpecifiedTextShadow {
+        fn from(shadow: Shadow) -> SpecifiedTextShadow {
```
Comment (emilio) [inline components/style/properties/longhand/inherited_text.mako.rs]: Given it's pretty much the same type, and they share parsing, the only thing that differs is the `to_computed_value` impl, right?

**Review by emilio — APPROVED** [review]
Looks reasonable to me, if you double-check they use the exact same parsing rules. If so, r=me with or without the nit below.

File: components/style/properties/longhand/inherited_text.mako.rs (hunk 2 above)
Comment (canaltinova) [inline components/style/properties/longhand/inherited_text.mako.rs]: Yes, also `Shadow` has 2 additional fields but they are not getting parsed if `disable_spread_and_inset` argument is true in parse method(which is true in that property).

File: components/style/properties/longhand/inherited_text.mako.rs (hunk 1 above)
Comment (canaltinova) [inline components/style/properties/longhand/inherited_text.mako.rs]: Yeah, that looks better.

**Comment by canaltinova** [comment]
I double-checked parsing methods and they are nearly same except third value rejects negative values in `Shadow`. Fortunately that's what we need :)

@bors-servo r=emilio

**Pull request merged**